### PR TITLE
Fix new empty array creation options default in Boolean filter

### PR DESCRIPTION
**Pull request by amcsi**
See zendframework/zend-servicemanager#159

### CODE REVIEW

File: src/Boolean.php [hunk 1]
```diff
@@ -61,7 +61,7 @@ class Boolean extends AbstractFilter
      */
     public function __construct($typeOrOptions = null, $casting = true, $translations = [])
     {
-        if ($typeOrOptions !== null) {
+        if ($typeOrOptions !== null && $typeOrOptions !== []) {
```
Comment (zf2timo) [inline src/Boolean.php]: I think a  `empty($typeOrOptions)` condition dose the same, but is easier to read.

**Comment by amcsi** [comment]
@zf2time I think that broke the unit tests where the class was constructed with a 0 intending to be the type

**Comment by zf2timo** [comment]
@amcsi yeah, you are right. LGTM.

**Comment by amcsi** [comment]
Alternative fix in other project: zendframework/zend-servicemanager#164

**Comment by amcsi** [comment]
Closing this, because zendframework/zend-servicemanager#164 makes this PR not needed anymore.

You should all upgrade to the latest Service Manager release of 2.7.8 https://github.com/zendframework/zend-servicemanager/releases/tag/release-2.7.8

**Pull request closed without merging**